ManageIQ/manageiq
# Handle images with multiple DVDs for SCVMM

This addresses https://bugzilla.redhat.com/show_bug.cgi?id=1284084

This treats DVD info as an array instead of a single object. It also reduces the data being sent over the wire a bit by selecting only those fields that we actually use, so there will be a slight performance improvement overall.

Steps for Testing/QA
--------------------

Add a new SCVMM infrastructure provider, or refresh an existing one using this code. Then check the device list for a VM that has one or more DVD's attached.

~~Why it's a WIP~~
~~--------------------~~

~~While this eliminates the bug, and inspection of the underlying array of hashes looks good, it doesn't actually appear to get saved to the guest_devices table and I'm not sure why yet.~~

~~On a side note, is there any reason we can't set :device_type to "DVD"? In the schema, it looks like a simple string field.~~


~~@bronaghs @jteehan Need some help please.~~
